Branch-SDK/Branch-SDK/Branch.m

```diff
@@ -886,12 +886,13 @@ - (void)applyReferralCode:(NSString *)code andCallback:(callbackWithParams)callb
 
 - (void)generateShortUrl:(NSArray *)tags andAlias:(NSString *)alias andType:(BranchLinkType)type andChannel:(NSString *)channel andFeature:(NSString *)feature andStage:(NSString *)stage andParams:(NSString *)params andCallback:(callbackWithUrl)callback {
     self.urlLoadCallback = callback;
-    dispatch_async(self.asyncQueue, ^{
-        BNCServerRequest *req = [[BNCServerRequest alloc] init];
-        req.tag = REQ_TAG_GET_CUSTOM_URL;
-        BNCLinkData *post = [self prepareLinkDataFor:tags andAlias:alias andType:type andChannel:channel andFeature:feature andStage:stage andParams:params];
-        
-        if (![self.linkCache objectForKey:post]) {
+    
+    BNCServerRequest *req = [[BNCServerRequest alloc] init];
+    req.tag = REQ_TAG_GET_CUSTOM_URL;
+    BNCLinkData *post = [self prepareLinkDataFor:tags andAlias:alias andType:type andChannel:channel andFeature:feature andStage:stage andParams:params];
+    
+    if (![self.linkCache objectForKey:post]) {
```

Comment:
Haha, you paranoid :-) Yup, I know this shorthand for NSDictionary, but self.linkCache here is actually a BNCLinkCache object.